File: odroid/Share/src/bert/share/message/HandlerType.java
/**
 * Copyright 2019. Charles Coughlin. All Rights Reserved.
 *                 MIT License.
 */
package bert.share.message;


/**
 * Recognized types of messsage handler. These are used to match specific
 * process instance with definitions in the configuration file. 
 * The "dispatcher" is a server application and accesses all controllers.
 */
public enum HandlerType
{
            COMMAND,
            REQUEST,
            SERIAL,
            TABLET,
            TERMINAL
            ;
          
 /**
  * @return  a comma-separated list of all handler types in a single String.
  */
  public static String names()
  {
    StringBuffer names = new StringBuffer();
    for (HandlerType type : HandlerType.values())
    {
      names.append(type.name()+", ");
    }
    return names.substring(0, names.length()-2);
  }
}

File: Bert/common/src/main/kotlin/chuckcoughlin/bert/common/message/HandlerType.kt
/**
 * Copyright 2019-2023. Charles Coughlin. All Rights Reserved.
 * MIT License.
 */
package chuckcoughlin.bert.common.message

/**
 * This is a marker for the Dispatcher to let it keep track of the controller
 * from which an individual message had come, so it can determine where it
 * needs to go.
 */
enum class HandlerType {
    BITBUCKET,
    COMMAND,
    DISPATCHER,
    INTERNAL,
    REQUEST,
    SERIAL,
    TABLET,
    TERMINAL,
    UNDEFINED;

    companion object {
        /**
         * @return  a comma-separated list of all handler types in a single String.
         */
        fun names(): String {
            val names = StringBuffer()
            for (type in values()) {
                names.append(type.name + ", ")
            }
            return names.substring(0, names.length - 2)
        }
    }
}